code/DDSCodeTester.cpp

```diff
@@ -3969,372 +3969,518 @@ class CustomizedDataReaderListener : public DataReaderListener
 
 void dds_qos_examples()
 {
+    // Taken out of the examples to avoid bloating them
+    DomainParticipantFactory* factory_ = DomainParticipantFactory::get_instance();
+    DomainParticipant* participant_ =
+            DomainParticipantFactory::get_instance()->create_participant(0, PARTICIPANT_QOS_DEFAULT);
+    Subscriber* subscriber_ =
+            participant_->create_subscriber(SUBSCRIBER_QOS_DEFAULT);
+    Publisher* publisher_ =
+            participant_->create_publisher(PUBLISHER_QOS_DEFAULT);
+    Topic* topic_ =
+            participant_->create_topic("TopicName", "DataTypeName", TOPIC_QOS_DEFAULT);
+    DataReader* reader_ = subscriber_->create_datareader(topic_, DATAREADER_QOS_DEFAULT);
+    DataWriter* writer_ = publisher_->create_datawriter(topic_, DATAWRITER_QOS_DEFAULT);
```

Comment:
I don't love this approach. I believe there will be leaks every time you overwrite a previous `DataWriter`. It'd make more sense to me for `writer_` to be a `unique_ptr` with a destructor that calls `publisher_.delete_datawriter();`.

Reply:
I would obviously suggest the same change to all the entities you overwrite, like `participant_` and `topic_`.

Reply:
These lines are never shown in the docs. It is just to make compiling code and avoid creating a new participant/publisher/writer or participant/subscriber/reader in each QoS example. I believe it makes the examples more readable and less repetitive, although it is not the best use of the API. We could consider to extend each example with the creation of the proper participant and/or publisher.